### PR TITLE
Bug fixes from PR 22

### PR DESCRIPTION
**Pull request by ksivaman**
Signed-off-by: Kirthi Shankar Sivamani <ksivamani@nvidia.com>

### CODE REVIEW

File: transformer_engine/pytorch/fp8.py [hunk 1]
```diff
@@ -314,7 +314,7 @@ def _default_get_amax(
     if amax_compute_algo == "max":
         amax = torch.max(amax_history, dim=0).values
     else:  # amax_compute_algo == "most_recent"
-        amax = amax_history[0]
+        amax = amax_history[0].clone()
```
Comment (ksivaman) [inline transformer_engine/pytorch/fp8.py:317]: In the `update_amax_history` call that follows, for the case with no amax history we avoid the roll kernel, but as a result we also overwrite this amax when calling `fill_` on the tensor. Making a copy avoids this.

**Comment by ptrendx** [comment]
pytest returns exit code 5 when no tests are collected (which is the case for `test_fp8.py` on non-H100 GPU with your PR -> this is interpreted as failure by the CI, which is why A100, TU102 and V100 tests failed. Could we bundle the `test_fp8.py` in the same run command with the other tests to avoid that?

**Comment by ksivaman** [comment]
Works now. The ci failure is due to the unrelated onnx tolerance @ptrendx

**Review by ptrendx — APPROVED** [review]

**Pull request merged**